relays/table_transform.ts

```diff
@@ -23,10 +24,10 @@ export default function(msg: Message): void|Message|Message[]|Promise<Message> {
 	let rounds = roundsArray(msg.data.stage.rounds);
 
 	let data = {
-		header: [NUMBER, NAME, TOP].concat(rounds),
-		data: msg.data.ranking.map(function(rank) {
+		header: [RANK, NUMBER, NAME, TOP].concat(rounds),
+		data: msg.data.ranking.map(function(rank, index) {

```

Comment:
Hmmm, good point on the generic bit!
I tend to agree that the transform could change things such that the (generic) display could handle it, e.g. with a 'null' value and a smart (page-wrap-aware) table display like you suggested.

If I understood you right, you mean something like (not tested...):
```ts
export default function(msg: Message): void|Message|Message[]|Promise<Message> {
	let rounds = roundsArray(msg.data.stage.rounds);
	let lastRank: string | null = null;
	let data = {
		header: [RANK, NUMBER, NAME, TOP].concat(rounds),
		data: msg.data.ranking.map(function(rank, index) {
			var scores = rank.scores.map((score) => score || 0);
			let displayRank = (lastRank !== rank.rank) ? rank.rank : null;
			lastRank = displayRank;
			return [displayRank, rank.team.number, rank.team.name, rank.highest].concat(scores);
		})
	};

	return new Message('table:setData', data);
}
```

And then the table display could decide to hide the value while it's on the same page, or repeat it when it's on the next, or always repeat it, or do other smart stuff. Yeah, I like it.

We probably need the change as you made it right now, though, given that the table display doesn't support that fancy mode yet, so I'll merge it :)